Source/WebCore/bindings/js/JSFileSystemWritableFileStreamCustom.cpp

```diff
@@ -0,0 +1,100 @@
+{
+    auto& vm = JSC::getVM(&lexicalGlobalObject);
+    auto* clientData = static_cast<JSVMClientData*>(vm.clientData);
+
+    JSC::MarkedArgumentBuffer arguments;
+    arguments.append(stream.internalWritableStream().guardedObject());
+    ASSERT(!arguments.hasOverflowed());
+    auto& writerPrivateName = clientData->builtinFunctions().writableStreamInternalsBuiltins().acquireWritableStreamDefaultWriterPrivateName();
+    auto writerOrException = InternalWritableStream::invokeWritableStreamFunction(lexicalGlobalObject, writerPrivateName, arguments);
+    if (writerOrException.hasException())
+        return { };
+
+    auto writer = writerOrException.releaseReturnValue();
+    arguments.clear();
+    arguments.append(writer);
+    arguments.append(data);
+    ASSERT(!arguments.hasOverflowed());
+    auto& writePrivateName = clientData->builtinFunctions().writableStreamInternalsBuiltins().writableStreamDefaultWriterWritePrivateName();
+    auto result = InternalWritableStream::invokeWritableStreamFunction(lexicalGlobalObject, writePrivateName, arguments);
+
+    arguments.clear();
+    arguments.append(writer);
+    auto& releasePrivateName = clientData->builtinFunctions().writableStreamInternalsBuiltins().writableStreamDefaultWriterReleasePrivateName();
+    InternalWritableStream::invokeWritableStreamFunction(lexicalGlobalObject, releasePrivateName, arguments);
+    if (result.hasException())
+        return { };
+
+    return result.releaseReturnValue();
+}
+
+JSC::JSValue JSFileSystemWritableFileStream::write(JSC::JSGlobalObject& lexicalGlobalObject, JSC::CallFrame& callFrame)
+{
+    return writeImpl(wrapped(), lexicalGlobalObject, callFrame.argument(0));
+}
+
+JSC::JSValue JSFileSystemWritableFileStream::seek(JSC::JSGlobalObject& lexicalGlobalObject, JSC::CallFrame& callFrame)
+{
+    std::optional<uint64_t> position;
+    {
+        auto scope = DECLARE_THROW_SCOPE(lexicalGlobalObject.vm());
+        auto positionConversionResult = convert<IDLUnsignedLong>(lexicalGlobalObject, callFrame.argument(0));
+        if (!positionConversionResult.hasException(scope))
+            position = positionConversionResult.returnValue();
+    }
+    FileSystemWritableFileStream::WriteParams params { FileSystemWriteCommandType::Seek, std::nullopt, position, std::nullopt };
+    auto data = toJS<IDLDictionary<FileSystemWritableFileStream::WriteParams>>(lexicalGlobalObject, *JSC::jsCast<JSDOMGlobalObject*>(&lexicalGlobalObject), params);
+    return writeImpl(wrapped(), lexicalGlobalObject, data);
+}
+
+JSC::JSValue JSFileSystemWritableFileStream::truncate(JSC::JSGlobalObject& lexicalGlobalObject, JSC::CallFrame& callFrame)
+{
+    std::optional<uint64_t> size;
+    {
+        auto scope = DECLARE_THROW_SCOPE(lexicalGlobalObject.vm());
+        auto sizeConversionResult = convert<IDLUnsignedLong>(lexicalGlobalObject, callFrame.argument(0));
+        if (!sizeConversionResult.hasException(scope))
+            size = sizeConversionResult.returnValue();
```

Comment:
If there is an exception, we should likely return early

Reply:
Same as above: I didn't return early here as it seems should go through "write a chunk" according to spec.